appserver/payara-appserver-modules/microprofile/rest-client/src/main/java/fish/payara/microprofile/restclient/RestClientServletContainerInitializerBlacklist.java

```diff
@@ -0,0 +1,80 @@
+ *
+ * When distributing the software, include this License Header Notice in each
+ * file and include the License file at glassfish/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * The Payara Foundation designates this particular file as subject to the "Classpath"
+ * exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+
+package fish.payara.microprofile.restclient;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+import fish.payara.web.loader.ServletContainerInitializerBlacklist;
+
+import javax.servlet.ServletContainerInitializer;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Extension point that omits {@link ServletContainerInitializer ServletContainerInitializers} annotated with
+ * {@link RegisterRestClient}. If we try and initialise these interfaces Jersey will fail them on validation (correctly)
+ * since you will end up with a duplicate mapping (the interface itself against the actual endpoint implementation).
+ *
+ * Using an extension mechanism rather than having this check directly in
+ * {@link org.glassfish.web.loader.ServletContainerInitializerUtil} allows us to avoid pulling MicroProfile APIs into
+ * the core server as a dependency.
+ *
+ * @author Andrew Pielage
+ */
+public class RestClientServletContainerInitializerBlacklist implements ServletContainerInitializerBlacklist {
+
+    @Override
+    public void removeServletContainerInitializers(Map<Class<? extends ServletContainerInitializer>,
+            Set<Class<?>>> initializerList) {
+
+        for (Map.Entry<Class<? extends ServletContainerInitializer>, Set<Class<?>>> e : initializerList.entrySet()) {
+            Class<? extends ServletContainerInitializer> initializer = e.getKey();
+
+            if (e.getValue() != null) {
+                // Don't try to initialise MicroProfile Rest Client interfaces as servlets
+                initializerList.put(initializer,
```

Comment:
A bit convoluted way to remove the classes in the existing lists, but I think it looks good. One major nitpick: The variable should be named `initializerMap` or `listedInitializersMap` instead.